src/ui/components/post/post_listing.rs

```diff
@@ -1,151 +1,308 @@
-use crate::ui::components::common::icon::{
-  Icon,
-  IconType::{Block, Crosspost, Downvote, Report, Save, Upvote, VerticalDots},
-};
-use lemmy_api_common::{
-  lemmy_db_schema::newtypes::*, //{PersonId, PostId},
-  lemmy_db_views::structs::*,   //PostView,
-  person::*,                    //{BlockPerson, BlockPersonResponse},
-  post::*, //{CreatePostLike, CreatePostReport, PostReportResponse, PostResponse, SavePost},
+use crate::{
+  errors::{LemmyAppError, LemmyAppErrorType},
+  lemmy_client::*,
+  ui::components::common::icon::{
+    Icon,
+    IconType::{Block, Comments, Crosspost, Downvote, Report, Save, Upvote, VerticalDots},
+  },
 };
+use lemmy_api_common::{lemmy_db_views::structs::*, person::*, post::*};
 use leptos::*;
 use leptos_router::*;
+use web_sys::SubmitEvent;
 
 #[server(VotePostFn, "/serverfn")]
-pub async fn vote_post_fn(post_id: i32, score: i16) -> Result<PostResponse, ServerFnError> {
-  use crate::lemmy_client::LemmyClient;
-  use actix_web::web;
-  use leptos_actix::extract;
+pub async fn vote_post_fn(post_id: i32, score: i16) -> Result<Option<PostResponse>, ServerFnError> {
+  use lemmy_api_common::lemmy_db_schema::newtypes::PostId;
 
   let form = CreatePostLike {
     post_id: PostId(post_id),
     score,
   };
+  let result = LemmyClient.like_post(form).await;
+
+  // use leptos_actix::redirect;
 
-  Ok(extract(|client: web::Data<awc::Client>| async move { client.like_post(form).await }).await??)
+  match result {
+    Ok(o) => Ok(Some(o)),
+    Err(_e) => {
+      // redirect(&format!("/?error={}", serde_json::to_string(&e)?)[..]);
+      Ok(None)
+    }
+  }
 }
 
 #[server(SavePostFn, "/serverfn")]
-pub async fn save_post_fn(post_id: i32, save: bool) -> Result<PostResponse, ServerFnError> {
-  use crate::lemmy_client::LemmyClient;
-  use actix_web::web;
-  use leptos_actix::extract;
+pub async fn save_post_fn(post_id: i32, save: bool) -> Result<Option<PostResponse>, ServerFnError> {
+  use lemmy_api_common::lemmy_db_schema::newtypes::PostId;
 
   let form = SavePost {
     post_id: PostId(post_id),
     save,
   };
+  let result = LemmyClient.save_post(form).await;
+
+  // use leptos_actix::redirect;
 
-  Ok(extract(|client: web::Data<awc::Client>| async move { client.save_post(form).await }).await??)
+  match result {
+    Ok(o) => Ok(Some(o)),
+    Err(_e) => {
+      // redirect(&format!("/?error={}", serde_json::to_string(&e)?)[..]);
+      Ok(None)
+    }
+  }
```

Comment:
Why are the returns for all of the server functions on this page returning `Result<Option<T>, ServerFnError>` instead of `Result<T, ServerFnError>`?

Reply:
i should have marked the PR as draft again before pushing update - this code may have been removed or changed